### ISSUE TITLE
TextArray not saving more than 1 element.

### ISSUE THREAD

**Issue by trentmillar** [keystonejs/keystone#3170]
I have a model using `TextArray`. It works fine unless I attempt to push more that one element prior to saving. It looks like only the last element is persisted on `save` and any additionals are lost.

**Comment by trentmillar** [comment]
I should have mentioned. This is all fine via the Admin UI and you can add notes no problem, however pushing strings onto the array programmatically is where this issue comes into play.

**Comment by mxstbr** [comment]
> This is all fine via the Admin UI and you can add notes no problem

Could you show some code? Since it works from the admin interfaced, I'm inclined to say this is a problem with your usage, maybe?

**Comment by christroutner** [comment]
I have also been experiencing this bug. Updating an element somewhere in the middle of the TextArray does not persist to the database when calling either save() or getUpdateHandler(). Neither calls persist the data to the database.

I can consistently reproduce this bug, but only in a very complex and proprietary piece of software I'm writing. I tried to reproduce it with this test script:

https://github.com/christroutner/keystone-textarray-test

But those two test scripts fail to reproduce the bug.

**Comment by christroutner** [comment]
I added a second and third test script to the keystone-textarray-test repo that build on top of the first and add complexity. 

I still have not been able to reproduce the bug with the test script, but I did experience it again on a totally separate piece of software from the first. Pretty frustrating!

What I've taken away from my test scripts lack of ability to reproduce the bug is that I need to use a REST APIs to interact with TextArrays() in order to avoid the bug that appears in my other software. That's what I plan to do moving forward since this bug only seems to appear when in more complex code using models with mixed field types and varying calls between save() and getUpdateHandler().
